extensions/data-transfer/portability-data-transfer-microsoft/src/main/java/org/dataportabilityproject/transfer/microsoft/transformer/TransformerService.java

```diff
@@ -0,0 +1,16 @@
+package org.dataportabilityproject.transfer.microsoft.transformer;
+
+/**
+ * Transforms instances into different types.
+ */
+public interface TransformerService {
+
+    /**
+     * Transform the input instance into an instance of a result type.
+     *
+     * @param resultType the type to transform to
+     * @param input the input instance
+     */
+    <T> TransformResult<T> transform(Class<T> resultType, Object input);
```

Comment:
Do we need Class<T> if we have generic T in the template?

Reply:
I think we do since the implementation needs to know what type is requested:

https://gist.github.com/jimmarino/3024ee7f1ab28f08c0d502b75820f9b6